### ISSUE TITLE
SERVER COMMUNICATION ERROR *FIX*

### ISSUE THREAD

**Issue by Abro61209** [Tustin/PlayStationDiscord#86]
Hey all, I wanted to share this with you guys because it seems I found a fix. I did this fix on a Mac computer, but I'm sure this fix is the same for Windows. 

What I did was I downloaded the 3.0.1 release. I signed in, it was working with discord, all was well. Then I decided "why not install the latest release and try again?" It turns out, when you drag it to the applications folder on a mac, it asks you to replace it. I did and it seemed to retain my cache data because I was automatically signed in with the latest release! 

**SO basically, install 3.0.1, sign in, and reinstall the latest release as it should keep you signed in. This should be possible on Windows because this program's "appdata" folder should be untouched.**

The only issue is (on mac at least) that there are no games icons on discord. Not sure if bug or not. Oh well, hope this helped!

**Comment by Tustin** [comment]
I think I’m going to try downgrading the version of Electron back to the version I used before 3.1.0 (when these issues started popping up). That’s honestly the only change I can see that might have broke the login. If I had to guess, one of the Electron updates added something to the window that causes Sony’s anti-bot protection to trigger.

**Comment by RMesado** [comment]
It worked!!!!

**Comment by RudeBoyEEEE** [comment]
Worked for me, too!

**Comment by seanbarney** [comment]
Worked here on Mac too!

**Comment by Andorem** [comment]
Worked for me on Windows 10! Thanks!

**Comment by akline25** [comment]
Where Can i get this fix? I'm having the same issue.

**Comment by Tustin** [comment]
I pushed a new version that downgrades Electron to the version I used in PlayStationDiscord 3.0.2 and below (which seemed to have some success for people). Please try it and let me know if you can login 😄. https://github.com/Tustin/PlayStationDiscord/releases/tag/v3.1.2

**Comment by hauyuw** [comment]
I downloaded v3.1.2, installed it on my Windows 10 desktop, and was FINALLY able to sign into PSN without the dreaded error. PS4 game registered on my discord and everything! Thanks for the fix!

**Comment by AeonX2** [comment]
Have just done a fresh install of v3.1.2 and am getting the login problem on Win 10

**Comment by Klairm** [comment]
Getting this problem also on Manjaro Linux 5.8
